js_modules/dagster-ui/packages/ui-core/src/assets/SavedZoomLevel.tsx

```diff
@@ -0,0 +1,31 @@
+import {MutableRefObject, useEffect} from 'react';
+
+import {SVGViewport} from '../graph/SVGViewport';
+import {getJSONForKey} from '../hooks/useStateWithStorage';
+
+const LINEAGE_GRAPH_ZOOM_LEVEL = 'lineageGraphZoomLevel';
+
+export const SVGSaveZoomLevel = ({scale}: {scale: number}) => {
+  useEffect(() => {
+    try {
+      window.localStorage.setItem(LINEAGE_GRAPH_ZOOM_LEVEL, JSON.stringify(scale));
+    } catch (err) {
+      // no-op
+    }
+  }, [scale]);
+  return <></>;
+};
+
+export function useLastSavedZoomLevel(
+  viewportEl: MutableRefObject<SVGViewport | undefined>,
+  layout: import('../asset-graph/layout').AssetGraphLayout | null,
+  assetGraphId: string,
+) {
+  useEffect(() => {
+    if (viewportEl.current && layout) {
+      const lastZoomLevel = Number(getJSONForKey(LINEAGE_GRAPH_ZOOM_LEVEL));
+      viewportEl.current.autocenter(false, lastZoomLevel);
+      viewportEl.current.focus();
+    }
+  }, [viewportEl, layout, assetGraphId]);
```

Comment:
You didn't actually use the assetGraphId here? I imagine the intent was to persist the zoom levels of different graphs independently?

Reply:
Ahh interesting... let me dig into this a bit, I think we may have been trying to make sure the zoom resets to the saved zoom when you select assets

Reply:
It looks like that was the case - i renamed this var so it's more obvious what it's doing here!